### PR TITLE
Add security, fossa, docs workflows; dependabot

### PR DESCRIPTION
**Pull request by sjberman**
New workflows for vulnerability and static analysis scanning, fossa uploads, and docs linting checks. Fixed up docs issues that were uncovered through scanning.

Added dependabot for updating go modules and github actions.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by avahahn — APPROVED** [review]

**Review by chase-kiefer — APPROVED** [review]
LGTM

**Review by lucacome — APPROVED** [review]

File: docs/content/tutorials/kic/ingress-udp-walkthrough.md [hunk 1]
```diff
@@ -143,24 +143,24 @@ To route UDP requests to an application in the mesh through the NGINX Plus Ingre
 
 Now that everything for the NGINX Plus Ingress Controller is deployed, we can now send datagrams to the udp-listener application.
 
-3. Use the IP and port defined in the [Install NGINX Plus Ingress Controller](#install-nginx-plus-ingress-controller) section to send a netcat UDP message:
+1. Use the IP and port defined in the [Install NGINX Plus Ingress Controller](#install-nginx-plus-ingress-controller) section to send a netcat UDP message:
```
Comment (jbyers19) [inline docs/content/tutorials/kic/ingress-udp-walkthrough.md:146]: Was restarting the numbered list intentional?

Comment (sjberman) [reply docs/content/tutorials/kic/ingress-udp-walkthrough.md:146]: Since they're separated by a header, the linter wants it to be a new list.

Comment (jbyers19) [reply docs/content/tutorials/kic/ingress-udp-walkthrough.md:146]: Cool. Just making sure.

**Review by jbyers19 — APPROVED** [review]

**Pull request merged**